lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -109,7 +92,7 @@ internal sealed class MapTaskHost<TMapInput, TMapOutput> : ITask, IObserver<IClo
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            while (Interlocked.Read(ref _shouldCloseTask) == 0)
+            while (!_taskCloseHandler.ShouldCloseTask())
```

Comment:
I am not sure why he put MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive() in using, does he want to dispose it after receiving each message? With this using in the middle, I cannot put the condition
mapInput.ControlMessage == MapControlMessage.Stop in the while statement unless assign it to another variable. What do you think?